WordPress/src/test/java/org/wordpress/android/ui/posts/PostUtilsUploadProcessingTest.kt

```diff
@@ -0,0 +1,98 @@
+package org.wordpress.android.ui.posts
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import org.assertj.core.api.Assertions
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.util.helpers.MediaFile
+
+@RunWith(MockitoJUnitRunner::class)
+class PostUtilsUploadProcessingTest {
+    private val siteUrl = "https://wordpress.org"
+    private val remoteImageUrl = "https://wordpress.org/gutenberg/files/2018/07/Screenshot-1-1.png"
+    private val remoteVideoUrl = "https://videos.files.wordpress.com/qeJFeNa2/macintosh-plus-floral-shoppe-02-e383aae382b5e38395e383a9e383b3e382af420-e78fbee4bba3e381aee382b3e383b3e38394e383a5e383bc-1_hd.mp4"
+    private val localMediaId = "112"
+    private val remoteMediaId = "97629"
+    private val attachmentPageUrl = "https://wordpress.org?p=97629"
+    private val oldImageBlock = """<!-- wp:image {"id":112,"align":"full"} -->
```

Comment:
Could you also add test cases where id is not the first attribute. Because there's no guarantee that id will be the first attribute. For example in media-text block, when we place the media at right and the text on left id becomes the second attribute. So I think we should add this case for all blocks.

Reply:
Indeed! I've added test cases for all blocks. It turns out, image was the only one failing this assertion, because the regex had a typo (now fixed). :smiley:

Reply:
wow, so it did help to add some new test cases. I am super happy we have unittests!

Reply:
Me too! Very helpful to have them, especially with regex changes.